### ISSUE TITLE
Serialization throws unexpected checked exception

### ISSUE THREAD

**Issue by ghost** [RuedigerMoeller/fast-serialization#178]
This program

```
public class Foo implements java.io.Serializable {
    public static void main(String[] args) throws java.io.IOException {
        new org.nustaq.serialization.FSTObjectOutput().writeObject(new Foo());
    }
    private void writeObject(java.io.ObjectOutputStream out) throws java.io.IOException {
        throw new java.io.IOException("Boom!");
    }
}
```

throws _java.lang.reflect.InvocationTargetException_ (instead of _java.io.IOException_). This is a checked exception, so I cannot reasonably catch it (unless I write something like "_catch Exception_" or "_catch Throwable_")

**Comment by RuedigerMoeller** [comment]
you might catch InvocationTargetException and check for the underlying reason. Its also possible to catch "Throwable", examine and rethrow. So this is a minor one. Theoretically fst could provide the Invocation-unwinding code and rethrow the original exception, however unsure wether this messes up the stack trace then

**Comment by ghost-0** [comment]
`try { } catch (InvocationTargetException e) { }`

is **illegal** if the code inside the try-block does not throw InvocationTargetException (because InvocationTargetException is a checked exception). So I must write something awkward like:

try {
    new org.nustaq.serialization.FSTObjectOutput().writeObject(new Foo());
} catch (Throwable t) {
    if (t instanceof InvocationTargetException) {
        Throwable cause = ((InvocationTargetException)t).getCause();
        throw cause instanceof IOException ? (IOException)cause : new Error("Surprise again", cause);
    }
    else rethrow(t);
}

static void rethrow(Throwable e) {  Test.<RuntimeException> doThrow0(e);  }
@SuppressWarnings("unchecked") static <E extends Throwable> void doThrow0(Exception e) throws E { throw (E) e; }

Another possibility is:

@SuppressWarnings("unused") // needed because the code throws InvocationTargetException even though this is not declared
void serialize() throws InvocationTargetException {
    new org.nustaq.serialization.FSTObjectOutput().writeObject(new Foo());
}

And the third possibility is using the standard Java serialization (which throws the original IOException without wrapping it in a checked exception)

**Comment by RuedigerMoeller** [comment]
fixed with 2.49, however only this specific place, there migth be more places where invocationtarget is reported though ...
